src/bootstrap/bootstrap.py

```diff
@@ -31,8 +31,16 @@ def get(url, path, verbose=False):
 
     try:
         download(sha_path, sha_url, verbose)
+        if os.path.exists(path):
+            if verify(path, sha_path, verbose):
+                print("using already-download file " + path)
+                return
+            else:
+                print("ignoring already-download file " + path + " due to failed verification")
+                os.unlink(path)
         download(temp_path, url, verbose)
-        verify(temp_path, sha_path, verbose)
+        if not verify(temp_path, sha_path, verbose):
```

Comment:
perhaps this should pass `verbose` as `True` unconditionally to ensure we always print out the mismatching hashes?

Similarly, the `verbose` above seems like it could always be `False` as we don't want to print out mis-verifications there.